array.c

```diff
@@ -1159,11 +1159,11 @@ rb_ary_initialize(int argc, VALUE *argv, VALUE ary)
 }
 
 /*
- * Returns a new array populated with the given objects.
+ * Returns a new +Array+ object, populated with the given objects:
+ *
+ *   Array.[]( 1, 'a', /^A/) # => [1, "a", /^A/]
+ *   Array.[]                # => []
```

Comment:
Can we also keep the example using `Array[1, 'a', /^A/]`? It's the same method call but using different syntax (the most idiomatic syntax).

Reply:
Restored (and moved the idiomatic syntax to top).